### PR TITLE
chore: 🤖 Upgrade addons/rose dependencies

### PR DESCRIPTION
**Pull request by dvonanderson**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-11587

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/JIRA_TICKET_NUMBER)

## Description
- Upgraded `addons/rose` dependencies to match all other addons.
- Ran yarn lint and fixed all errors including: `no-curly-component-invocation`, `no-passed-in-event-handlers` 
- Added template lint rule for `rose` addons to ignore prettier linting. Per conversations with a  Packer cloud team member , the also like a more readable approach to the render `hbs` in testing so they ignore `prettier` linting entirely,

## Checklist:

~~- [ ] I have added before and after screenshots for UI changes~~
~~- [ ] I have added JSON response output for API changes~~
~~- [ ] I have added steps to reproduce and test for bug fixes in the description~~
~~- [ ] I have commented on my code, particularly in hard-to-understand areas~~
- [x] My changes generate no new warnings
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~

### CODE REVIEW

File: addons/rose/tests/dummy/config/ember-try.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        npm: {
+          devDependencies: {
+            'ember-source': await getChannelURL('canary'),
+          },
+        },
+      },
+      embroiderSafe(),
+      embroiderOptimized(),
+    ],
+  };
```
Comment (lisbet-alvarez) [inline addons/rose/tests/dummy/config/ember-try.js:57]: non-blocking, for understanding: from what I read online, this is used to test using different ember versions, how were the specific versions in this file chosen?

Comment (lisbet-alvarez) [reply addons/rose/tests/dummy/config/ember-try.js:57]: +1 to adding as techtime topic. I think its ok to leave this file in for now

File: addons/rose/tests/integration/components/rose/button-test.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ module('Integration | Component | rose/button', function (hooks) {
   });
 
   test('it is type="submit" when @submit={{true}}', async function (assert) {
```
Comment (lisbet-alvarez) [inline addons/rose/tests/integration/components/rose/button-test.js:24]:
```suggestion
  test('it is type="submit" when @onSubmit={{true}}', async function (assert) {
```

**Review by lisbet-alvarez — APPROVED** [review]
🙌

File: addons/rose/tests/dummy/config/ember-try.js (hunk 1 above)
Comment (calcaide) [reply addons/rose/tests/dummy/config/ember-try.js:57]: Good question @lisbet-alvarez I learned about `ember-try` last year when performing Ember upgrades. +1 to add this topic to tech time, just for educational purposes 😄 

P.S: it is a know pattern your ember apps and specially addons to have and use `ember-try`.

File: addons/rose/.template-lintrc.js [hunk 3]
```diff
@@ -31,5 +31,11 @@ module.exports = {
         'no-passed-in-event-handlers': false,
       },
     },
+    {
+      files: '**/tests/**/*.{js,gts,gjs}',
```
Comment (calcaide) [inline addons/rose/.template-lintrc.js:35]: None blocking question: this line, was add after running the command to update ember or you add it manually?

**Review by calcaide — APPROVED** [review]
LGTM!! Awesome work with the upgrade @dvonanderson

File: addons/rose/.template-lintrc.js (hunk 3 above)
Comment (dvonanderson) [reply addons/rose/.template-lintrc.js:35]: @calcaide I added manually.  I think after the discussion with you and the team about formatting in our test files, I reached out to the cloud team and this is what they currently do.  I mean we probably don't need the `.gts` format, but I hope that we will get to `.gjs` someday

File: addons/rose/tests/integration/components/rose/dropdown/button-test.js [hunk 4]
```diff
@@ -22,7 +22,7 @@ module('Integration | Component | rose/dropdown/button', function (hooks) {
   });
 
   test('it is type="submit" when @submit={{true}}', async function (assert) {
```
[the remark this answers is not in the document]
Comment (dvonanderson) [reply addons/rose/tests/integration/components/rose/dropdown/button-test.js:24]: yes good catch thanks @DhariniJeeva

File: addons/rose/addon/components/rose/button/index.hbs [hunk 5]
```diff
@@ -5,7 +5,7 @@
 
 <button
   ...attributes
-  type={{if @submit 'submit' 'button'}}
+  type={{if @onSubmit 'submit' 'button'}}
```
Comment (DhariniJeeva) [inline addons/rose/addon/components/rose/button/index.hbs:8]: Why are we changing the argument to `onSubmit`?

Comment (dvonanderson) [reply addons/rose/addon/components/rose/button/index.hbs:8]: @DhariniJeeva `no-passed-in-event-handlers` lint error from `ember-template-lint`

**Review by DhariniJeeva — APPROVED** [review]
awesome work! ty for the upgrade.

**Pull request merged**